src/components/features/PairwiseSearch/FeatureInfoPlot.vue

```diff
@@ -0,0 +1,220 @@
+          xaxis: {
+            title: "Sources",
+            showticklabels: false,
+            categoryorder: "array",
+          },
+          yaxis: {
+            title: "Number of " + "<br>" + "Pairwise Features",
+          },
+          hoverlabel: {
+            font: {
+              size: 10,
+            }
+          },
+          modebar: {
+            displayModeBar: true,
+            left: 0.1,
+          }
+        };
+      },
+
+      plotData: function() {
+        return this.createPlotData();
+      }
+    },
+
+    created() {
+      this.getRelationshipsForGenes();
+    },
+  
+    methods: {
+      createPlotData() {
+          let dataDesc2Type = new Map();
+          if(this.dataDescriptions){
+            for(let dataDescription of this.dataDescriptions){
+              let dataType = dataDescription.dataType;
+              if(dataType === "Gene_Coexpression"){
+                let dataResource = dataDescription.id.split("|")[0];
+                dataType += "_" + dataResource;
+              }
+
+              dataDesc2Type.set(dataDescription.id, dataType);
+            }
+          }
+
+          let dataType2pairwiseRelationship = new Map();
+          if(this.results){
+            for(let result of this.results){
+              let descType = dataDesc2Type.get(result.dataDesc.id);
+              if(descType === undefined) descType = "unknown";
+
+              let geneRelationship = dataType2pairwiseRelationship.get(descType);
+              if(geneRelationship === undefined){
+                geneRelationship = {relationshipType: [], posNum: [], text: []};
+                dataType2pairwiseRelationship.set(descType, geneRelationship);
+              }
+
+                geneRelationship.relationshipType.push(result.dataDesc.id);
+                //TODO: case where no posNum
+                let posNum = result.posNum ? result.posNum: 0;
+                let negNum = result.negNum ? result.negNum: 0;
+                geneRelationship.posNum.push(posNum + negNum);
+                geneRelationship.text.push(this.generateText(result.posNum, result.dataDesc.id));
+            }
+          }
+
+          let plotData = [];
+          let relationshipTypes = dataType2pairwiseRelationship.keys();
+          for(let relationshipType of relationshipTypes) {
+            let pairwiseData = dataType2pairwiseRelationship.get(relationshipType);
+            let name = relationshipType;
+
+            let relationshipGeneData = {
+              name: name,
+              x: pairwiseData.relationshipType,
+              y: pairwiseData.posNum,
+              text: pairwiseData.text,
+              type: "scattergl",
+              mode: "markers",
+              marker: {
+                size: 5,
+              }
+            }
+            plotData.push(relationshipGeneData);
+          }
+          return plotData;
+        },
+  
+        generateText(numFeatures, relationship) {
+          if(relationship){
```

Comment:
Better to do something like this:

if (relationships === undefined) return

to keep your code clean.